File: src/RealtimeServer/scriptureforge/models/text-data.ts
import { Canon } from '@sillsdev/scripture';
import { DeltaOperation } from 'rich-text';

export const TEXTS_COLLECTION = 'texts';
export const TEXT_INDEX_PATHS: string[] = [];

export type TextType = 'source' | 'target';

export function getTextDocId(projectId: string, book: number, chapter: number, textType: TextType = 'target'): string {
  return `${projectId}:${Canon.bookNumberToId(book)}:${chapter}:${textType}`;
}

export interface TextData {
  ops?: DeltaOperation[];
}

File: src/SIL.XForge.Scripture/Models/TextData.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using SIL.Scripture;
using SIL.XForge.Models;
using SIL.XForge.Realtime.RichText;

namespace SIL.XForge.Scripture.Models
{
    public class TextData : Delta, IIdentifiable
    {
        public static string GetTextDocId(string projectId, int book, int chapter)
        {
            return $"{projectId}:{Canon.BookNumberToId(book)}:{chapter}:target";
        }

        public TextData() { }

        public TextData(Delta delta) : base(delta) { }

        [JsonIgnore]
        public string Id { get; set; }

        [JsonIgnore]
        public Dictionary<string, object> ExtraElements { get; set; }
    }
}
